tests/Unit/Framework/MockDistributedObject.hpp

```diff
@@ -129,6 +138,54 @@ struct LocalCoreId {
   size_t value;
 };
 
+inline bool operator==(const LocalCoreId& lhs,
+                       const LocalCoreId& rhs) noexcept {
+  return lhs.value == rhs.value;
+}
+
+inline bool operator!=(const LocalCoreId& lhs,
+                       const LocalCoreId& rhs) noexcept {
+  return not(lhs==rhs);
+}
+
+/// Wraps a size_t representing the global core number.
```

Comment:
Could you add a few words here and to `LocalCoreId` explaining their difference? It's reasonably obvious (I assume "local" identifies the core in the node and "global" identifies the core among all nodes), but it would be good to make that clear. Generally, I don't like docstrings that just repeat the symbol name without CamelCase 😄